azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/model/WebAppWindowsRuntime.java

```diff
@@ -0,0 +1,167 @@
+    public static final WebAppWindowsRuntime TOMCAT10_JAVA11 = new WebAppWindowsRuntime("Tomcat 10.0", "Java 11");
+    public static final WebAppWindowsRuntime TOMCAT10_JAVA8 = new WebAppWindowsRuntime("Tomcat 10.0", "Java 8");
+    public static final WebAppWindowsRuntime TOMCAT9_JAVA17 = new WebAppWindowsRuntime("Tomcat 9.0", "Java 17");
+    public static final WebAppWindowsRuntime TOMCAT9_JAVA11 = new WebAppWindowsRuntime("Tomcat 9.0", "Java 11");
+    public static final WebAppWindowsRuntime TOMCAT9_JAVA8 = new WebAppWindowsRuntime("Tomcat 9.0", "Java 8");
+    public static final WebAppWindowsRuntime TOMCAT85_JAVA17 = new WebAppWindowsRuntime("Tomcat 8.5", "Java 17");
+    public static final WebAppWindowsRuntime TOMCAT85_JAVA11 = new WebAppWindowsRuntime("Tomcat 8.5", "Java 11");
+    public static final WebAppWindowsRuntime TOMCAT85_JAVA8 = new WebAppWindowsRuntime("Tomcat 8.5", "Java 8");
+
+    private static final AtomicReference<Boolean> loaded = new AtomicReference<>(Boolean.FALSE);
+    private static final LinkedHashSet<WebAppWindowsRuntime> RUNTIMES = Sets.newLinkedHashSet(Arrays.asList(
+        JAVASE_JAVA17, JAVASE_JAVA11, JAVASE_JAVA8,
+        TOMCAT10_JAVA17, TOMCAT10_JAVA11, TOMCAT10_JAVA8,
+        TOMCAT9_JAVA17, TOMCAT9_JAVA11, TOMCAT9_JAVA8,
+        TOMCAT85_JAVA17, TOMCAT85_JAVA11, TOMCAT85_JAVA8
+    ));
+
+    @Getter
+    @EqualsAndHashCode.Include
+    private final OperatingSystem operatingSystem = OperatingSystem.WINDOWS;
+    /**
+     * container name in upper case, e.g. 'JAVA', 'TOMCAT', 'JBOSS', 'JETTY'
+     */
+    @Getter
+    @EqualsAndHashCode.Include
+    private final String containerName;
+    /**
+     * container version number, e.g. '17', '17.0.4', '1.8.202'
+     */
+    @Getter
+    @EqualsAndHashCode.Include
+    private final String containerVersionNumber;
+    /**
+     * java version number, e.g. '17', '17.0.4', '1.8.0_202', '1.8.0_202_ZULU'
+     */
+    @Getter
+    @EqualsAndHashCode.Include
+    private final String javaVersionNumber;
+    private final String javaVersionDisplayText;
+    @Getter
+    private final boolean deprecatedOrHidden;
+
+    private WebAppWindowsRuntime(@Nonnull WebAppMinorVersion webContainer, @Nonnull WebAppMinorVersion javaMinorVersion) {
+        final WindowsJavaContainerSettings containerSettings = webContainer.stackSettings().windowsContainerSettings();
+        final WebAppRuntimeSettings javaSettings = javaMinorVersion.stackSettings().windowsRuntimeSettings();
+        this.deprecatedOrHidden = BooleanUtils.isTrue(containerSettings.isHidden())
+            || BooleanUtils.isTrue(javaSettings.isHidden())
+            || BooleanUtils.isTrue(containerSettings.isDeprecated())
+            || BooleanUtils.isTrue(javaSettings.isDeprecated());
+        this.containerName = containerSettings.javaContainer().toUpperCase();
+        this.containerVersionNumber = containerSettings.javaContainerVersion().toUpperCase();
```

Comment:
why we need upper case container version

Reply:
> why we need upper case container version

keep consistency, not sure if there are versions like `xxx_ZULU_xxx`